tests/unit/directives.spec.ts

```diff
@@ -43,10 +43,31 @@ describe('v-track-click directives', (): void => {
     console.error = jest.fn();
     wrapper.find('div').trigger('click');
 
-    expect(pluginOptions.trackClickHandler).toHaveBeenCalledWith(testCase.eventName, testCase.eventProps);
+    expect(pluginOptions.trackClickHandler).toHaveBeenCalledWith(testCase.eventName, { foo: 'bar' });
     testCase.eventProps.foo = 'car';
     await Vue.nextTick();
 
-    expect(pluginOptions.trackClickHandler).toHaveBeenCalledWith(testCase.eventName, testCase.eventProps);
+    expect(pluginOptions.trackClickHandler).toHaveBeenCalledWith(testCase.eventName, { foo: 'car' });
+  });
+
+  it('tracks an event on click even if element is removed from the DOM', async (): Promise<void> => {
+    const testCase = {
+      template: `<div v-if="eventProps.foo == 'bar'" v-track-click @click="eventProps.foo = 'car'" event-name="kamehameha" :event-props="eventProps"></div>`,
+      eventName: 'kamehameha',
+      eventProps: { foo: 'bar' },
+    };
+
+    const component = Vue.extend({
+      template: testCase.template,
+      data: () => ({ eventProps: testCase.eventProps }),
+    });
+    const wrapper = shallowMount(component, { localVue });
+    expect(pluginOptions.trackClickHandler).not.toHaveBeenCalled();
+
+    console.error = jest.fn();
+    wrapper.find('div').trigger('click');
+
+    await Vue.nextTick();
+    expect(pluginOptions.trackClickHandler).toHaveBeenCalledWith(testCase.eventName, { foo: 'car' });
```

Comment:
so the test case is useful because line 55 changes eventProps.foo == "car".  

Your test cases checks that the @click has run first (thus the `foo: car`) but that the track click handler is still run. 

Cool this makes sense.

Reply:
A little bit more than that. Notice the v-if in the test template, it actually removes the element from the DOM when the @click runs. So if you remove the setTimeout call from the unbind, this test starts to fail.